build-system/tasks/visual-diff.rb

```diff
@@ -78,12 +81,82 @@ def waitForWebServer()
   until isWebServerRunning()
     sleep(1)
     tries += 1
-    break if tries > 15
+    break if tries > WEBSERVER_TIMEOUT_SECS
   end
   isWebServerRunning()
 end
 
 
+# Checks the current status of a Percy build.
+#
+# Args:
+# - buildId: ID of the ongoing Percy build.
+# Returns:
+# - The full response from the build status server.
+def getBuildStatus(buildId)
+  statusUri = URI("#{BUILD_STATUS_URL}?build_id=#{buildId}")
+  Net::HTTP.start(statusUri.host, statusUri.port, use_ssl: true) do |https|
+    request = Net::HTTP::Get.new(statusUri)
+    response = https.request(request)
+    if response.code == "200"
+      buildStatus = JSON.parse(response.body)
+    end
+  end
+rescue SystemCallError
+  # Fail gracefully, and wait for the next attempt.
+  return
+end
+
+
+# Waits for Percy to finish processing a build.
+#
+# Args:
+# - buildId: ID of the ongoing Percy build.
+# Returns:
+# - True if build is complete.
+def waitForBuildCompletion(buildId)
+  tries = 0
+  until ['finished', 'failed'].include?(getBuildStatus(buildId)['state'])
+    sleep(5)
+    tries += 1
+    break if tries > (BUILD_PROCESSING_TIMEOUT_SECS / 5)
+  end
+end
+
+
+# Verifies that a Percy build succeeded and didn't contain any visual diffs.
+#
+# Args:
+# - buildId: ID of the Percy build.
+def verifyBuildStatus(buildId)
+  status = getBuildStatus(buildId)
```

Comment:
Since you always call `waitForBuildCompletion` first, maybe pass in the status that the wait process yielded?